shared/@hedvig-ui/Input/input.tsx

```diff
@@ -1,66 +1,239 @@
+  inputSize?: InputSize
+  affix?: boolean
+}>`
+  ${({ inputSize, affix }) => iconStyles(inputSize, affix)}
+  color: ${({ theme }) => theme.success};
+`
+
+const ErrorIcon = styled(ExclamationCircleFill)<{
+  inputSize?: InputSize
+  affix?: boolean
+}>`
+  ${({ inputSize, affix }) => iconStyles(inputSize, affix)}
+  color: ${({ theme }) => theme.danger};
+`
+
+const LoadingIcon = styled.div<{
+  inputSize?: InputSize
+  affix?: boolean
+}>`
+  ${({ inputSize, affix }) => iconStyles(inputSize, affix)}
+  display: flex;
+  align-items: center;
+  justify-content: center;
 
+  &:after {
+    content: ' ';
+    display: block;
+    width: 90%;
+    height: 90%;
+    border-radius: 50%;
+    border: 3px solid #000;
+    border-color: #000 transparent #000 transparent;
+    animation: ${rotate} 1s linear infinite;
   }
 `
 
-export const Input: React.FC<CustomInputProps> = (props) => (
-  <StyledSemanticInput
-    {...props}
-    onKeyDown={(e) => {
-      if (shouldIgnoreInput(e.key)) {
-        e.preventDefault()
-        return
-      }
-
-      if (e.keyCode === Keys.Escape.code) {
-        e.preventDefault()
-        e.target.blur()
-        return
-      }
-
-      if (props.onKeyDown) {
-        props.onKeyDown(e)
-      }
-    }}
-  />
-)
+const CustomIcon = styled.div`
+  position: absolute;
+  left: 20px;
+
+  width: 20px;
+  height: 20px;
+
+  display: flex;
+  align-items: center;
+  justify-content: center;
+
+  & * {
+    width: 100%;
+    height: 100%;
+  }
+`
+
+const Affix = styled.div<{ inputSize?: InputSize }>`
+  font-size: 12px;
+
+  display: flex;
+  align-items: center;
+  justify-content: center;
+
+  position: absolute;
+  right: ${({ inputSize }) => (inputSize === 'small' ? '3px' : '5px')};
+
+  border-radius: 0.25rem;
+  background-color: ${({ theme }) => theme.accentBackground};
+  border: 1px solid ${({ theme }) => theme.border};
+
+  overflow: hidden;
+  white-space: nowrap;
+  text-overflow: ellipsis;
+
+  height: calc(
+    100% - ${({ inputSize }) => (inputSize === 'small' ? '6px' : '10px')}
+  );
+  width: 50px;
+`
+
+interface AffixType {
+  content: string
+}
+
+export interface InputProps {
+  autoFocus?: boolean
+  disabled?: boolean
+  error?: boolean
+  success?: boolean
+  muted?: boolean
+  placeholder?: string
+  value?: string | number
+  id?: string
+  type?: string
+  name?: string
+  onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void
+  onKeyDown?: (event: React.KeyboardEvent<HTMLInputElement>) => void
+  ref?: React.RefObject<HTMLInputElement>
+  onFocus?: (event: React.ChangeEvent<HTMLInputElement>) => void
+  onBlur?: (event: React.ChangeEvent<HTMLInputElement>) => void
+  style?: React.CSSProperties
+  size?: InputSize
+  loading?: boolean
+  icon?: React.ReactNode
+  affix?: AffixType
+}
+
+export const Input: React.FC<InputProps> = ({
+  success,
+  error,
+  disabled,
+  loading,
+  muted,
+  icon,
+  size: inputSize,
+  affix,
+  style,
+  ...props
+}) => {
```

Comment:
Instead of trying to cover everything explicilty, could we perhaps combine with existing props for native HTML elements, e.g. something like:

```suggestion
export interface InputProps {
  autoFocus?: boolean
  disabled?: boolean
  error?: boolean
  success?: boolean
  muted?: boolean
  placeholder?: string
  value?: string | number
  size?: InputSize
  loading?: boolean
  icon?: React.ReactNode
  affix?: AffixType
}

export const Input: React.FC<InputProps & InputHTMLAttributes<HTMLInputElement>> = ({
  success,
  error,
  disabled,
  loading,
  muted,
  icon,
  size: inputSize,
  affix,
  style,
  ...props
}) => {
```

(not sure if it's exactly like this, but you get the point 😄 )